catalyst-gateway/bin/src/db/index/block/rbac509/mod.rs

```diff
@@ -155,6 +111,11 @@ impl Rbac509InsertQuery {
                     &report,
                 ));
             },
+            Err(RbacCacheAddError {
+                catalyst_id: None, ..
```

Comment:
It seems for me makes sense to make a separate error variant under the `RbacCacheAddError` (meaning make `RbacCacheAddError` as enum and distinguish between different scenarios.